### PR TITLE
rpm: update to 4.16.0.

### PR DESCRIPTION
**Pull request by glaulher**

### CODE REVIEW

File: srcpkgs/rpm/template [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+makedepends="libgcrypt-devel binutils-devel db-devel elfutils-devel file-devel libarchive-devel
  libcap-devel lua52-devel nss-devel popt-devel python3-devel libgomp-devel"
 short_desc="Red Hat Package Management Utils"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://rpm.org"
 changelog="https://rpm.org/wiki/Releases/${version}.html"
-distfiles="http://ftp.rpm.org/releases/rpm-4.15.x/rpm-${version}.tar.bz2"
-checksum=ddef45f9601cd12042edfc9b6e37efcca32814e1e0f4bb8682d08144a3e2d230
+distfiles="http://ftp.rpm.org/releases/rpm-4.16.x/rpm-${version}.tar.bz2"
```
Comment (ericonr) [inline srcpkgs/rpm/template:17]: `rpm-${version%.*}.x` could work?

**Comment by ericonr** [comment]
Please fix the `xlint` error as well, and I think the musl build is missing `#include <fcntl.h>` from the `rpmdb.c` file.

**Comment by glaulher** [comment]
I accidentally deleted the branch I was testing on, and I can't reset the pr, the xlint fix, how can I include the #include <fcntl.h> in the rpmdb.c file?

**Comment by ericonr** [comment]
You will probably need to open a new PR.

- For xlint, check the python version is indeed correct and remove the comment
- For fcntl, create a patch that adds the `#include <fcntl.h>` line to the include block in the file; I can try and get you a patch to do that.

**Comment by glaulher** [comment]
yesterday I can fix the xlint it's ok now, but I can't solve the question of the path

**Comment by ericonr** [comment]
Can you show the error messages this time? Remember you have to clean the build `./xbps-src clean rpm` for the patch to be applied.

**Comment by glaulher** [comment]
=> rpm-4.16.0_1: verifying checksum for distfile 'rpm-4.16.0.tar.bz2'... OK.
=> rpm-4.16.0_1: running do-extract hook: 00-distfiles ...
=> rpm-4.16.0_1: extracting distfile(s), please wait...
=> rpm-4.16.0_1: running do-patch hook: 00-patches ...
=> rpm-4.16.0_1: patching: fix-musl-rpmdb.patch.
=> ERROR: rpm-4.16.0_1: do-patch_00-patches: 'patch -sl ${_args} -i ${_patch} 2> /dev/null' exited with 2
=> ERROR:   in _process_patch() at common/hooks/do-patch/00-patches.sh:34
=> ERROR:   in hook() at common/hooks/do-patch/00-patches.sh:51
=> ERROR:   in run_func() at common/xbps-src/shutils/common.sh:21
=> ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:245
=> ERROR:   in run_step() at common/xbps-src/shutils/common.sh:71
=> ERROR:   in main() at common/xbps-src/libexec/xbps-src-dopatch.sh:33

**Comment by ericonr** [comment]
Take a look at the other patch, you might be missing part of the path... It has:

```
--- rpmio/digest_nss.c
+++ rpmio/digest_nss.c
```

**Comment by glaulher** [comment]
unfortunately I don't know how to go about correcting the error. help-me

**Comment by ericonr** [comment]
Open a new PR with your current attempt, I can help you better from there.

**Comment by glaulher** [comment]
ok, thanks

**Pull request closed without merging**